plugins/generator-1.21.1/neoforge-1.21.1/templates/gui/gui_window.java.ftl

```diff
@@ -74,6 +72,29 @@ public class ${name}Screen extends AbstractContainerScreen<${name}Menu> {
 		this.imageHeight = ${data.height};
 	}
 
+	public void onMenuStateUpdate(int elementType, String name, Object elementState) {
+		<#if data.getComponentsOfType("TextField")?has_content>
+		if (elementType == 0 && elementState instanceof String stringState) {
+			<#list data.getComponentsOfType("TextField") as component>
+				if (name.equals("${component.getName()}")) {
+					${component.getName()}.setValue(stringState);
+				}
+			</#list>
+		}
+		</#if>
+		<#if data.getComponentsOfType("Checkbox")?has_content>
+		if (elementType == 1 && elementState instanceof Boolean logicState) {
+			this.updateLock = true;

```

Comment:
Since currently there is no procedure block to set the checkbox value, only to get it, this part can be omitted from this PR, just leave a FTL comment:

`<#-- onMenuStateUpdate is not implemented for checkboxes, as there is no procedure block to set checkbox state currently -->`